src/anemoi/inference/checkpoint/__init__.py

```diff
@@ -63,3 +65,98 @@ def operational_config(self):
 
         LOG.warning("No operational configuration found. Using default configuration.")
         return {}
+
+    def validate_environment(
+        self,
+        all_packages: bool = False,
+        on_difference: Literal["warn", "error"] = "warn",
+    ) -> int:
+        """
+        Validate environment of the checkpoint against the current environment.
+
+        Parameters
+        ----------
+        all_packages : bool, optional
+            Check all packages in environment or just `anemoi`'s, by default False
+        on_difference : Literal['warn', 'error'], optional
+            What to do on difference, by default "warn"
+
+        Returns
+        -------
+        int
+            0 if environment is valid, 1 otherwise
+
+        Raises
+        ------
+        RuntimeError
+            If found difference and `on_difference` is 'error'
+        ValueError
+            If `on_difference` is not 'warn' or 'error'
+        """
+        train_environment = self.provenance_training
+        inference_environment = gather_provenance_info(full=False)
+
+        invalid_messages = {
+            "python": [],
+            "missing": [],
+            "mismatch": [],
+            "uncommitted": [],
+        }
+
+        if train_environment["python"] != inference_environment["python"]:
+            invalid_messages["python"].append(
+                f"Python version mismatch: {train_environment['python']} != {inference_environment['python']}"
+            )
+
+        for module in train_environment["module_versions"].keys():
+            if not all_packages and "anemoi" not in module:
+                continue
+
+            if module not in inference_environment["module_versions"]:
+                invalid_messages["missing"].append(f"Missing module in inference environment: {module}")
+            elif train_environment["module_versions"][module] != inference_environment["module_versions"][module]:
```

Comment:
For most modules it will be fine if version inference > training. So I would split this case up with different log messages so it's immediately clear if it's a problem or not:
- inference > training : a gentle reminder, but probably not a problem
- inference < training : danger, this is a problem